src/3.x/testing.md

```diff
@@ -47,4 +47,32 @@ $browser->waitUntilTrueForDuration('true', 120, 0.5);
 
 ## Continuous Integration
 
-If you would like to run the tests with a CI tool like Bitbucket Pipelines or Github Actions, you need to ensure that there is a Magento environment that Rapidez can work with. You could use a test environment for this, so you don't have to set up a Magento environment in Docker. However, you will need to create an SSH tunnel for the database connection, and for every run, orders will be created. If you have made it this far and you're interested, [ask us on Slack](https://rapidez.io/slack)!
\ No newline at end of file
+If you would like to run the tests with a CI tool like Bitbucket Pipelines or Github Actions, you need to ensure that there is a Magento environment that Rapidez can work with. You could use a test environment for this, so you don't have to set up a Magento environment in Docker. However, you will need to create an SSH tunnel for the database connection, and for every run, orders will be created. If you have made it this far and you're interested, [ask us on Slack](https://rapidez.io/slack)!
+
+## Static analysis
+
+Rapidez comes with PHPStan for static analysis (`composer analyse`) and Pint (`composer style` and `composer fix-style`) for code style analysis installed by default. These can be used in automated pipelines to help keep code quality under control.
+
+### GitHub actions
+
+The `rapidez/rapidez` repository contains an example PHPStan workflow [here](https://github.com/rapidez/rapidez/blob/master/.github/workflows/analyse.yml). Replace `composer analyse` with `composer style` to run Pint.
+
+### Bitbucket pipelines
+
+If you're using Bitbucket you can set up these pipelines with a php docker container. A pipeline step to run PHPStan could look something like this:
+
+```yaml
+    - step: &analyse
+        name: Static analysis
+        image:
+          name: $DOCKER_IMAGE
+          username: $DOCKER_USERNAME
+          password: $DOCKER_PASSWORD
```

Comment:
Maybe we could add a default here instead of variables, like:
```suggestion
        image: ghcr.io/justbetter/docker-ci-php-node:1-22.04-8.3-22
```